astropy/time/formats.py

```diff
@@ -269,6 +269,9 @@ def to_value(self, parent=None):
     def value(self):
         raise NotImplementedError
 
+    def _check_val_empty(self, val):
```

Comment:
I'm not sure this routine is needed (see later comment).

Also, it tests, I think, the wrong thing: an array with `shape=(2, 0)` is not the same as an array with empty shape `(0,)`. I also thought we can count on `val` being an `ndarray` here.